src/main/java/org/alfresco/deployment/sample/activemq/CamelRouteConfig.java

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright 2018 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.alfresco.deployment.sample.activemq;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+import org.apache.camel.RoutesBuilder;
+import org.apache.camel.builder.RouteBuilder;
+import org.apache.camel.component.amqp.AMQPComponent;
+import org.apache.camel.component.jms.JmsConfiguration;
+import org.apache.qpid.jms.JmsConnectionFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.jms.connection.CachingConnectionFactory;
+import org.springframework.retry.backoff.FixedBackOffPolicy;
+import org.springframework.retry.policy.SimpleRetryPolicy;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Jamal Kaabi-Mofrad
+ */
+@Configuration
+@ConditionalOnProperty(prefix = "messaging.config", name = "enabled", havingValue = "true")
+public class CamelRouteConfig
```

Comment:
This is the type of class we'd like to see in an Events SDK in the very near future.

Perhaps we should pull them into a package that would make it easier to move when we do that, or even go so far as to make it a separate Maven module or even separate repo now?  (that may be more work than we sized for this story though)

Reply:
I guess it is outside of this sprint's scope now. I purposely encapsulated all the required logic in one class so we can move it easily.